reference-implementation/test/transform-stream.js

```diff
@@ -207,67 +187,59 @@ test('TransformStream: by default, closing the writable waits for transforms to
   setTimeout(() => {
     t.equal(rsClosed, false, 'readable is not closed after a tick');
 
-    ts.writable.closed.then(() => {
-      t.equal(ts.writable.state, 'closed', 'writable becomes closed eventually');
+    writer.closed.then(() => {
+      // TODO: Is this expectation correct?
       t.equal(rsClosed, true, 'readable is closed at that point');
     })
     .catch(e => t.error(e));
   }, 0);
 });
 
 test('TransformStream: by default, closing the writable closes the readable after sync enqueues and async done', t => {
-  t.plan(3);
-
   const ts = new TransformStream({
-    transform(chunk, enqueue, done) {
+    transform(chunk, done, enqueue) {
       enqueue('x');
       enqueue('y');
       setTimeout(done, 50);
     }
   });
 
-  ts.writable.write('a');
-  ts.writable.close();
-  t.equal(ts.writable.state, 'closing', 'writable is closing');
-
-  ts.writable.closed.then(() => {
-    t.equal(ts.writable.state, 'closed', 'writable becomes closed eventually');
+  const writer = ts.writable.getWriter();
+  writer.write('a');
+  writer.close();
 
+  writer.closed.then(() => {
     return readableStreamToArray(ts.readable).then(chunks => {
       t.deepEqual(chunks, ['x', 'y'], 'both enqueued chunks can be read from the readable');
+      t.end();
     });
   })
   .catch(e => t.error(e));
 });
 
 test('TransformStream: by default, closing the writable closes the readable after async enqueues and async done', t => {
-  t.plan(3);
-
   const ts = new TransformStream({
-    transform(chunk, enqueue, done) {
+    transform(chunk, done, enqueue) {
```

Comment:
Yeah, this is what I should clearly explain. I thought that like streams' controller, it might be good to pass the TransformStream controlling functions at the time of creation so that we allow the implementation to start writing some header that is independent from what will be written to the writable side. Then, we'll introduce start(). start() would take function to enqueue(), close() and error(). They don't need to be given to each of transform() and flush() but passed just as a shortcut. But, chunk and done() are specific to each transform() call. So, I thought it would be good to regroup them like this.